### PR TITLE
Enum support

### PR DESCRIPTION
**Pull request by Yomanz**
Changes the type generation to use typescript `enum` rather than the custom type.

### CODE REVIEW

File: packages/schema/src/Codegen/files/generated/types.ts [hunk 1]
```diff
@@ -110,9 +113,21 @@ export class TypesFile extends File {
...
+              this.schemaTypeComments(type)
+            )}export enum ${type.name} { \n
+            ${(type as SchemaEnumType).enumValues.map(k => `${k} = '${k}' \n`)}
+          }`
+        )
+        .join('\n')}
 
       ${Object.values(this.codegen.schema.types)
-        .filter(type => type.kind !== 'INPUT_OBJECT')
+        .filter(type => !['INPUT_OBJECT', 'ENUM'].includes(type.kind))
```
Comment (samdenty) [inline packages/schema/src/Codegen/files/generated/types.ts:130]: can you keep this how it was, and just check for type.kind in the .map below?

```ts
          type =>
            `${this.generateComments(
              this.schemaTypeComments(type)
            )}${type.kind === 'enum' ? `export enum ${type.name} { \n
            ${(type as SchemaEnumType).enumValues.map(k => `${k} = '${k}' \n`)}
          }` : ORIG}`
```

File: packages/schema/src/Codegen/files/generated/types.ts [hunk 2]
```diff
@@ -98,6 +99,8 @@ export class TypesFile extends File {
       .filter(Boolean)
       .join('\n\n')
 
+    console.log(this.codegen.schema.types)
```
Comment (samdenty) [inline packages/schema/src/Codegen/files/generated/types.ts:102]: unnecessary console log

File: .changeset/slimy-bananas-remain.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'gqless': patch
```
Comment (samdenty) [inline .changeset/slimy-bananas-remain.md:2]: only need to release schema, can remove this line

File: packages/schema/src/Codegen/files/generated/types.ts (hunk 2 above)
Comment (Yomanz) [reply packages/schema/src/Codegen/files/generated/types.ts]: Removed

File: .changeset/slimy-bananas-remain.md (hunk 3 above)
Comment (Yomanz) [reply .changeset/slimy-bananas-remain.md:2]: Fixed

File: packages/schema/src/Codegen/files/generated/types.ts (hunk 1 above)
Comment (Yomanz) [reply packages/schema/src/Codegen/files/generated/types.ts]: Changed.

**Pull request merged**